package.json

```diff
@@ -39,6 +37,7 @@
     "istanbul": "0.4.4",
     "mocha": "3.0.2",
     "mocha-lcov-reporter": "1.2.0",
-    "sinon": "1.17.5"
+    "sinon": "1.17.5",
+    "bluebird": "^3.4.0"
```

Comment:
I used `bluebird` because the serverless source code and most plugin developers use this.  In any case I understand if you have preferences.

I've refactored to use `Q`, and updated the `devDependencies` to use `"q": "1.4.1"`